### PR TITLE
fix #100 bug when response is a string

### LINKED ISSUE #100: Bug when parsing a response that is not a serialized json
[This line](https://github.com/bigcommerce/bigcommerce-api-ruby/blob/master/lib/bigcommerce/exception.rb#L44) will raise JSON::ParserError if the env.body is a string.

### PR DESCRIPTION
**Pull request by gregory**
screenshot 2015-04-28 16 00 02

Fix: We should rescue the JSON parsing

cc @mattolson @pedelman

### CODE REVIEW

**Comment by gregory** [comment]
will add couple more things hold on before merging.

File: lib/bigcommerce/exception.rb [hunk 1]
```diff
@@ -41,11 +41,18 @@ module HttpErrors
 
     def throw_http_exception!(code, env)
       return unless ERRORS.keys.include? code
-      error_hash = env.body.empty? ? {} : JSON.parse(env.body, symbolize_names: true)
+      response_headers = {}
+      unless env.body.empty?
+        begin
```
Comment (mattolson) [inline lib/bigcommerce/exception.rb]: Do you mean to say `response_headers = begin`... ?

Comment (gregory) [inline lib/bigcommerce/exception.rb]: yes :)

**Comment by mattolson** [comment]
:+1:

**Pull request merged**